nativelib/src/main/scala/scala/scalanative/native/pure.scala

```diff
@@ -0,0 +1,10 @@
+package scala.scalanative
+package native
+
+/** Let optimizer assume that given method is pure.
+  * This means that repetitive calls to the same method
+  * with the same arguments are allowed to be eliminated.
+  * Additionally, if method result to given method is not used
+  * then the call can be elided altogether.
+  */
+final class pure extends scala.annotation.StaticAnnotation
```

Comment:
> we need some short-term solution to clean up of extraneous runtime module loads and runtime calls. 

Are you sure you do? As in, do you have benchmarks that show that applications are significantly slower if you do not use `@pure`?

> If it is in fact side-effecting in observable ways, it's programmer's fault.

i.e., undefined behavior. UB is fine, but it should be possible to *check* them in development mode. Here, I don't think one can check that `@pure` was not erroneously used (even at run-time).